src/indexer/segment_writer.rs

```diff
@@ -317,12 +318,13 @@ impl SegmentWriter {
 
 // This method is used as a trick to workaround the borrow checker
 fn write(
+    doc_opstamps: &Vec<Opstamp>,
```

Comment:
`&Vec<_>` is almost never a good idea. `&[Opstamp]` would have been better, but here I think we can pass `Vec<Opstamp>` and hence avoid the copy.

Reply:
Yeah, my first iteration was to move the Vec, but I saw that write was also being used in a &self method.

Reply:
Yes. I ended up doing a bit of refactoring.